banyan/resource_accesstier.go

```diff
@@ -23,11 +24,31 @@ func resourceAccessTier() *schema.Resource {
 		DeleteContext: resourceAccessTierDelete,
 		Schema:        AccessTierSchema(),
 		Importer: &schema.ResourceImporter{
-			StateContext: schema.ImportStatePassthroughContext,
+			StateContext: resourceImporter,
 		},
 	}
 }
 
+func resourceImporter(ctx context.Context, data *schema.ResourceData, i interface{}) ([]*schema.ResourceData, error) {
+	inID := data.Id()
+	parts := strings.Split(inID, ":")
+	if len(parts) == 1 {
+		data.SetId(inID)
+		return []*schema.ResourceData{data}, nil
+	}
+	if len(parts) != 2 || strings.ToLower(parts[0]) != "name" {
+		return nil, fmt.Errorf("invalid ID (%s), expected name:<name of access tier to import>", parts)
+	}
+	name := parts[1]
+	accessTierClient := i.(*client.Holder)
```

Comment:
would it be good to check if the type assertion succeeded?

Reply:
yes, sure shall add.